src/Validation/ScannableURLsRestController.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * ScannableURLsRestController class.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.2
+ */
+
+namespace AmpProject\AmpWP\Validation;
+
+use AMP_Validated_URL_Post_Type;
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Error;
+use WP_Post;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * Rest endpoint for fetching the scannable URLs.
+ *
+ * @since 2.2
+ * @internal
+ */
+final class ScannableURLsRestController extends WP_REST_Controller implements Delayed, Service, Registerable {
+
+	/** @var ScannableURLProvider ScannableURLProvider instance. */
+	private $scannable_url_provider;
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+		return 'rest_api_init';
+	}
+
+	/**
+	 * Constructor.
+	 *
+	 * @param ScannableURLProvider $scannable_url_provider Instance of ScannableURLProvider class.
+	 */
+	public function __construct( ScannableURLProvider $scannable_url_provider ) {
+		$this->namespace              = 'amp/v1';
+		$this->rest_base              = 'scannable-urls';
+		$this->scannable_url_provider = $scannable_url_provider;
+	}
+
+	/**
+	 * Registers all routes for the controller.
+	 */
+	public function register() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_items' ],
+					'args'                => [],
+					'permission_callback' => static function () {
+						return current_user_can( \AMP_Validation_Manager::VALIDATE_CAPABILITY );
```

Comment:
Please add a `use AMP_Validation_Manager` above.
```suggestion
						return current_user_can( AMP_Validation_Manager::VALIDATE_CAPABILITY );
```

Reply:
Thanks for that. Addressed in 7f41352